### PR TITLE
Improve help text for URL settings

### PR DESCRIPTION
**Pull request by sabernhardt**
Combines patches from [Trac 50886](https://core.trac.wordpress.org/ticket/50886) for General Settings page

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-admin/options-general.php [hunk 1]
```diff
@@ -25,14 +25,21 @@
...
-		'<p>' . __( 'If you want site visitors to be able to register themselves, as opposed to by the site administrator, check the membership box. A default user role can be set for all new users, whether self-registered or registered by the site admin.' ) . '</p>';
+	$options_help .= '<p>' . __( 'Two terms you will want to know are the WordPress URL and the site URL. The WordPress URL is where the core WordPress installation files are, and the site URL is the address a visitor uses in the browser to go to your site.' ) . '</p>' .
+		'<p>' . __( 'Though the terms refer to two different concepts, in practice, they can be the same address or different. For example, you can have the WordPress core files in the root directory (<code>https://example.com</code>), in which case the two URLs would be the same. Or the WordPress files can be in a subdirectory (<code>https://example.com/wordpress</code>). In that case, the site URL and the WordPress URL would be different.' ) . '</p>' .
```
Comment (audrasjb) [inline src/wp-admin/options-general.php]: Should add a link to this [HelpHub article](https://wordpress.org/documentation/article/giving-wordpress-its-own-directory/) somewhere?

**Comment by audrasjb** [comment]
The patch looks good to me. Only wondering whether it would be useful to point people to [this HelpHub link](https://wordpress.org/documentation/article/giving-wordpress-its-own-directory/)?

**Comment by sabernhardt** [comment]
The HelpHub link was already directly below the Site Address field, but it's probably worth including in both places.

**Comment by audrasjb** [comment]
Ah you're right. In that case, I'm not sure it's worth having it on both locations

File: src/wp-admin/options-general.php (hunk 1 above)
Comment (sabernhardt) [reply src/wp-admin/options-general.php]: _If_ we add the link, we'll need to update it. I used the old `support` URL instead of `documentation`.

**Comment by sabernhardt** [comment]
I updated the documentation URL. I'm more inclined to add the link, even if people might ignore it, in case someone does want it there. If the link should be removed, though, reverting the last two commits would be easy enough.

**Comment by audrasjb** [comment]
Thinking about it, I think we should put that link back in the help tabs anyway, because it's useful to have it in that context too 👍

File: src/wp-admin/options-general.php [hunk 2]
```diff
@@ -25,14 +25,25 @@
...
-		'<p>' . __( 'If you want site visitors to be able to register themselves, as opposed to by the site administrator, check the membership box. A default user role can be set for all new users, whether self-registered or registered by the site admin.' ) . '</p>';
+	$options_help .= '<p>' . __( 'Two terms you will want to know are the WordPress URL and the site URL. The WordPress URL is where the core WordPress installation files are, and the site URL is the address a visitor uses in the browser to go to your site.' ) . '</p>' .
+		'<p>' . sprintf(
+			/* translators: %s: Documentation URL. */
+			__( 'Though the terms refer to two different concepts, in practice, they can be the same address or different. For example, you can have the WordPress core files in the root directory (<code>https://example.com</code>), in which case the two URLs would be the same. Or the <a href="%s">WordPress files can be in a subdirectory</a> (<code>https://example.com/wordpress</code>). In that case, the site URL and the WordPress URL would be different.' ),
```
Comment (costdev) [inline src/wp-admin/options-general.php]:
```suggestion
			__( 'Though the terms refer to two different concepts, in practice, they can be the same address or different. For example, you can have the core WordPress installation files in the root directory (<code>https://example.com</code>), in which case the two URLs would be the same. Or the <a href="%s">WordPress files can be in a subdirectory</a> (<code>https://example.com/wordpress</code>). In that case, the WordPress URL and the site URL would be different.' ),
```
Nitpicks:
- "WordPress core files" is referred to as "core WordPress installation files" in the previous paragraph.
- "WordPress URL" comes before "site URL" in the previous paragraph.

File: src/wp-admin/options-general.php [hunk 3]
```diff
@@ -25,14 +25,25 @@
...
+			__( 'https://wordpress.org/documentation/article/giving-wordpress-its-own-directory/' )
+		) . '</p>' .
+		'<p>' . sprintf(
+			/* translators: 1: http://, 2: https:// */
+			__( 'Both site URLs and WordPress URLs can start with either %1$s or %2$s. A URL starting with %2$s will also show a padlock next to the address in the browser address bar. Both of those things signal to visitors that your site meets some basic security requirements, which can build trust with your users and with search engines.' ),
```
Comment (costdev) [inline src/wp-admin/options-general.php]:
```suggestion
			__( 'Both WordPress URLs and site URLs can start with either %1$s or %2$s. A URL starting with %2$s will also show a padlock next to the address in the browser address bar. Both of those things signal to visitors that your site meets some basic security requirements, which can build trust with your users and with search engines.' ),
```
Nitpick: "WordPress URL" comes before "site URL" in the previous paragraph.

**Review by costdev — CHANGES_REQUESTED** [review]
Thanks for the PR @sabernhardt! I've just left two nitpicks for consistent language 🙂

File: src/wp-admin/options-general.php [hunk 4]
```diff
@@ -25,14 +25,25 @@
...
+			'<code>http://</code>',
+			'<code>https://</code>'
+		) . '</p>' .
+		'<p>' . __( 'If you want site visitors to be able to register themselves, check the membership box. If you want the site administrator to register every new user, leave the box unchecked. In both cases, you can set a default user role for all new users, whether they register themselves or a site admin registers them.' ) . '</p>';
```
Comment (costdev) [inline src/wp-admin/options-general.php]: > In both cases ... whether case A or case B

This looks a bit tautological here. The cases have just been explained in the previous sentence.

**Suggested option 1**
Reduce repetition by removing from "whether they register themselves..." onwards.
```suggestion
		'<p>' . __( 'If you want site visitors to be able to register themselves, check the membership box. If you want the site administrator to register every new user, leave the box unchecked. In both cases, you can set a default user role for all new users.' ) . '</p>';
```
 
**Suggested option 2**
Remove "In both cases", and don't shorten "site administrator" to "site admin" after previously referring to "site administrator".
```suggestion
		'<p>' . __( 'If you want site visitors to be able to register themselves, check the membership box. If you want the site administrator to register every new user, leave the box unchecked. You can set a default user role for all new users, whether they register themselves or a site administrator registers them.' ) . '</p>';
```

Comment (sabernhardt) [reply src/wp-admin/options-general.php]: I suppose I prefer option 1, which could start with "In either case" to avoid using the word "both" again.

Comment (costdev) [reply src/wp-admin/options-general.php]: That works for me! 🙂

**Review by costdev — APPROVED** [review]
Thanks for the updates @sabernhardt! LGTM :+1:

**Review by SergeyBiryukov — CHANGES_REQUESTED** [review]
> Both WordPress URLs and site URLs can start with either `http://` or `https://`. A URL starting with `https://` will also show a padlock next to the address in the browser address bar. Both of those things signal to visitors that your site meets some basic security requirements, which can build trust with your users and with search engines.

I think this part can be confusing, as it does not mention that `https://` only works if you set up an SSL certificate, or if the hosting provider does that for you. I've seen people on support forums with this exact issue, changing these URLs to `https://` _before_ setting up SSL and then asking why the site can no longer be opened.

The suggested text seems to encourage changing the URLs to `https://` without explaining the consequences.

Should we try to clarify that here?

**Comment by costdev** [comment]
@SergeyBiryukov How does this sound?

> Both WordPress URLs and site URLs can start with either `http://` or `https://`. A URL starting with `https://` requires an SSL certificate, so be sure that you have one before changing to `https://`. With `https://`, a padlock will appear next to the address in the browser address bar. Both `https://` and the padlock signal that your site meets some basic security requirements, which can build trust with search engines and your visitors.

**Comment by audrasjb** [comment]
Sounds great to me 👍

**Comment by sabernhardt** [comment]
The proposed text is good, but I specifically want visitors before search engines in the last sentence.

I updated the PR in case it's worth committing this late in the cycle. However, with soft string freeze today, this ticket probably should be moved to 6.3.

**Review by SergeyBiryukov — APPROVED** [review]
Looks good to me 👍

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! Merged in [r55452](https://core.trac.wordpress.org/changeset/55452).

**Pull request closed without merging**